File: services/wallet-service/src/middleware/internalAuth.ts
import type { RequestHandler } from 'express';
import { env } from '../config/env';
import { UnauthorizedError } from './errorHandler';

export const internalAuth: RequestHandler = (req, _res, next) => {
  const secret = req.headers['x-internal-secret'];

  if (!secret || secret !== env.INTERNAL_SECRET) {
    return next(new UnauthorizedError('Invalid or missing internal secret'));
  }

  next();
};

File: services/ai-service/app/api/middleware/internal_auth.py
from fastapi import Header

from app.core.config import get_settings
from app.core.errors import UnauthorizedError


async def require_internal_secret(x_internal_secret: str | None = Header(default=None)) -> None:
    if x_internal_secret is None or x_internal_secret != get_settings().internal_secret:
        raise UnauthorizedError("Invalid or missing internal secret")
